### DISCUSSION TITLE
Support for Python 3.9?

### DISCUSSION THREAD

**Discussion by jrobble** [openmpf/openmpf#1546] [Q&A] [answered]
**Repost from Google Groups:**

Hello!

I am trying to implement Deepface into openMPF but I am unable to do so because openMPF requires its components to run python version 3.8.2 and deepface requires version 3.9. Do you know of any way around this when building the component through Docker or any general guidance? Whenever we try to import deepface in our component, we get an error of

```
An error occurred while running the job: ModuleNotFoundError: No module named '_bz2'
At:
  /usr/local/lib/python3.8/bz2.py(19): <module>
```

Thanks for any sort of help and let me know if more information needs to be provided!

Chad

**Comment by jrobble** [comment, marked as answer]
Hi Chad, are you using the OpenMPF 6.x Docker images? Those are based off the master branches.

Those images use CentOS 7, and we had to build Python 3.8 ourselves since it wasn't available via `yum`. By the end of the month we hope to release OpenMPF 7.0. In that release we're transitioning from CentOS 7 to using Ubuntu 20.04 as the base OS for all of our images. It comes with Python 3.8 baked in.

One developer recently encountered a "No module named '_lzma' " error when using the 6.x Python component base Docker image that went away once he transitioned to using the Docker image generated from develop. The error you're encountering seems very similar and is likely due to the fact that we didn't compile Python 3.8 with that compression library.

Our develop branch uses `apt-get` to install Python 3.8 [here](https://github.com/openmpf/openmpf-docker/blob/develop/components/python/Dockerfile#L47). That line could be updated to use Python 3.9, but I'm not sure what the impact would be on the OpenMPF Python SDK. You may want to try using Python 3.8 and see it it works with your library.

We currently don't have the develop Docker images posted to DockerHub. You could try [building them yourself](https://github.com/openmpf/openmpf-docker/blob/develop/README.md#pull-or-build-the-openmpf-docker-images). Otherwise, @brosenberg42 may be able to post beta versions next week.

**Reply by chadm8** [reply]
Thanks for the quick response Jeff!

Yes, we are using OpenMPF 6.x Docker images. Our team agrees that since the compression library was not compiled in Python3.8, that is resulting in our error with the module not being found. We think that we will just wait for the release of OpenMPF 7.0 and hope that Python 3.8 being baked in will solve our issue. We appreciate your help and hope that the transition to OpenMPF 7.0 goes smoothly!

**Reply by zramsis** [reply]
Hey I just wanted to check is there a status on the release for OpenMPF 7.0?

**Reply by jrobble** [reply]
The big features for the release have already landed to develop. It's just administrivia to get it packaged up and released. We're hoping to get it out on GitHub and Docker Hub by mid next week, if not sooner.

**Reply by jrobble** [reply]
We just released OpenMPF 7.0.0. The new base images are on Docker Hub. Please let us know if you have any issues.

**Comment by chadm8** [comment]
Awesome, thanks for letting me know!! Just ran our function we were struggling with and it now runs perfectly!
________________________________
